binlog_streamer.go

```diff
@@ -49,6 +55,7 @@ type BinlogStreamer struct {
 
 	logger         *logrus.Entry
 	eventListeners []func([]DMLEvent) error
+	eventHandlers  map[string]func(*replication.BinlogEvent, []byte, *BinlogEventState) ([]byte, error)
```

Comment:
Wonder if we can create an enum of "supported" events that you could create handlers for, and then use that enum as the map key. Would be slightly more robust than string keys at the cost of some extra code to maintain.

Reply:
I've cleaned this up a bit, and we have some bounded validations around the type of events you can attach a handler to.